### PR TITLE
BE request controllers #6

### PR DESCRIPTION
**Pull request by lipob**
### What this PR does (required):
- Issue #6 
- Request controllers GET, POST and PUT ready.
- GET : list of requests for logged in user
- POST : Create a new request
- UPDATE : Update request with approved or decline

### CODE REVIEW

File: server/controllers/request.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+const User = require("../models/User");
+const asyncHandler = require("express-async-handler");
+
+// @route GET /request
+// @desc get requests related to a logged user
+// @access Private
+exports.getRequests = asyncHandler( async (req, res, next) => {
+  const loggedUserId = req.user.id;
+  if (!loggedUserId) {
+    res.status(401);
```
Comment (moffatethan) [inline server/controllers/request.js:11]: We have middleware available that does this check for us already, I would suggest taking this logic out

File: server/controllers/request.js [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+      { [requestState]: true },
+      { new: true },
+      function(err, doc) {
+        if (err) {
+          res.status(400).send("Request not found");
+        } else {
+          res.send(doc);
+        }
+      }
+    )
```
Comment (moffatethan) [inline server/controllers/request.js:74]: We should use async/await here to be consistent with the project codebase.

**Review by moffatethan — COMMENTED** [review]
Left you some comments, thank you!

File: server/controllers/request.js (hunk 1 above)
Comment (lipob) [reply server/controllers/request.js]: FIXED. Should I add the "protect" middleware to Request controllers?

File: server/controllers/request.js (hunk 2 above)
Comment (lipob) [reply server/controllers/request.js]: FIXED

**Review by bonnieli — APPROVED** [review]

**Pull request merged**